src/Extensions/Hosting.Services.Web/Middlewares/ObsoleteCorrelationHeadersMiddleware.cs

```diff
@@ -89,6 +95,25 @@ private void ExtractCorrelationFromRequest(HttpRequest request)
 				? value.FirstOrDefault(s => !string.IsNullOrWhiteSpace(s))
 				: null;
 
+		private static string? ExtractHeader(IHeaderDictionary headers, IEnumerable<string> names)
+		{
+			foreach (string name in names)
+			{
+				string? value = ExtractHeader(headers, name);
+				if (!string.IsNullOrWhiteSpace(value))
```

Comment:
You are checking twice for nullorwhitespace. You could add an out parameter to the function and return true when it is not null?
```suggestion
				bool notNull = ExtractHeader(headers, name, out string? value);
				if (notNull)
				{
```

Reply:
Resolving in favour of another comment, the `ExtractHeader` function was never needed since IHeaderDictionary will return String.Empty if the key is not present.

https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.http.iheaderdictionary.item?view=aspnetcore-3.1#Microsoft_AspNetCore_Http_IHeaderDictionary_Item_System_String_